python/az/aro/azext_aro/custom.py

```diff
@@ -129,10 +130,18 @@ def aro_list_credentials(client, resource_group_name, resource_name):
 
 def aro_update(client, resource_group_name, resource_name, worker_count=None,
                no_wait=False):
+    current = client.get(resource_group_name, resource_name)
```

Comment:
There is a potential for a race condition when two users (or CI jobs or whatever) run the command at about the same time:

* User 1 reads and modifies the data
* User 2 reads and modifies the data in a different way to User 1
* User 2 sends a patch
* User 1 sends a patch

Client side race condition should be handled by this:

https://github.com/Azure/ARO-RP/blob/51b85f3ef4fc6ffba1195a98cfe4dbd11d97aea7/pkg/frontend/openshiftcluster_putorpatch.go#L80-L83

So one of the users will get an error "Request is not allowed in provisioningState 'Updating'."

But I have a feeling there is also a race condition on the server where two workers receive a patch request, validate  `ProvisioningState` which looks fine for both, but then one of them does it's job faster than another one...

Reply:
> But I have a feeling there is also a race condition on the server where two workers receive a patch request, validate ProvisioningState which looks fine for both, but then one of them does it's job faster than another one...

That should definitely not be the case.  On the server side, all patch/update operations should fail if the etag of the submitted (modified) document does not match the etag of the database document (optimistic concurrency).  So the racing operations must be serialised correctly, and one will get the "Request is not allowed in provisioningState 'Updating'." error.